### PR TITLE
Fix formations generator

### PR DESCRIPTION
**Pull request by burner1024**
## Description
make_formations.py is outdated and doesn't match the corresponding 2da's, see #1759.

Hope you don't mind conversion to python 3.
Files are generated with 20 positions.
Files generated with updated script match current 2das exactly, with exception of GATHER formation. I couldn't figure out the algo used to calculate the current one, if there even was any. But first 6 positions match exactly, and the rest shouldn't matter according to [this](https://github.com/gemrb/gemrb/issues/1759#issuecomment-1338761973).

This is the new gather, for reference.
6-new

## Checklist

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] I used the same coding style as the surrounding code 
- [x] I have tested the proposed changes
- [ ] I extended the documentation, if necessary
- [ ] The proposed change builds also on our build bots (check after submission)

### CODE REVIEW

File: admin/make_formation.py [hunk 1]
```diff
@@ -314,6 +315,15 @@ def generate_wedge():
...
 	print()
 
+def generate_none():
+	print("NONE", end=' ')
+	end = " "
+	for i in range(num_coords):
+		if i == (num_coords - 1):
+			end = ""
+		print("0 0", end=end)
+	print()
```
Comment (lynxlynxlynx) [inline admin/make_formation.py:325]: Why not just use something like `print("NONE ", "0 0 " * (num_coords - 1),  "0 0\n")`

File: admin/make_formation.py [hunk 2]
```diff
@@ -77,34 +77,31 @@ def generate_t():
...
-				yloc = 24 * i // 3
+		if i == 0:
+			xloc = 0 # centre
+			yloc = -36 # front
+		elif i == 1:
+			xloc = 48 # right
+			yloc = -24
+		elif i == 2:
+			xloc = -48 # left
+			yloc = -24
```
Comment (lynxlynxlynx) [inline admin/make_formation.py:95]: this block is redundant

**Comment by lynxlynxlynx** [comment]
Thanks! And yes, a python3 port is welcome. 
It all started with you complaining about formation shape, but these descriptions and code all look like they preserve the existing state. Is the shape fix coming later, did I misunderstand something?

File: admin/make_formation.py (hunk 1 above)
Comment (burner1024) [reply admin/make_formation.py]: That doesn't generate the same line, but I fixed it.

File: admin/make_formation.py (hunk 2 above)
Comment (burner1024) [reply admin/make_formation.py:95]: All 1-3 are at negative y, and all the rest are at positive.

**Comment by burner1024** [comment]
Yes, first I wanted to get the exact state, then make changes.
I think formations are mostly good, except the pentacle one.

**Comment by lynxlynxlynx** [comment]
The same files are generated here as well. Please also fix generate_formation_files.sh though, it should be using unhardcoded instead of override. And ideally it would inspect $0 to find out where it is and derive the path prefix from there.

**Comment by burner1024** [comment]
Fixed shell script, and improved pentacle.
Comparison:
5-2-old
5-new

**Comment by bradallred** [comment]
you should also commit the generated formato.2da files

**Comment by burner1024** [comment]
Converted to draft for now, will revisit the shapes once there's a working projection fix.

**Comment by burner1024** [comment]
Ok, I made Protect and Gather match bg2 more closely, screenshots attached.

Do you want me to review other formations?

1o
1g
2o
2g

**Comment by lynxlynxlynx** [comment]
Thanks and please do.

**Comment by burner1024** [comment]
OK, I adjusted the rest, mostly the difference was gemrb variant placing characters closer together than originals.
<details> 
  <summary>Screenshots, originals and updated GemRB</summary>

wedge_o
wedge_g
tria_o
tria_g
t_o
t_g
s_o
s_g
rank_o
rank_g
line_o
line_g
3b2_o
3b2_g
2b3_o
2b3_g

</details>

**Comment by lynxlynxlynx** [comment]
Thanks!

**Pull request merged**